### PR TITLE
Feature/bc 1071 finish/archive & restore task action

### PR DESCRIPTION
**Pull request by odalys-dataport**
# Short Description

Closing and opening of tasks over the 3-dot-menu

## Links to Ticket and related Pull-Requests

https://ticketsystem.hpi-schul-cloud.org/browse/BC-1071

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/store/tasks.ts [hunk 1]
```diff
@@ -71,6 +73,31 @@ export class TaskModule extends VuexModule {
...
+				archived: task.archived,
+			});
+
+			await this.fetchAllTasks();
+			if (FinishedTasksModule.isInitialized) {
+				await FinishedTasksModule.refetchTasks();
+			}
+
+			this.setStatus("completed");
+		} catch (error) {
```
Comment (SevenWaysDP) [inline src/store/tasks.ts:95]: this is a problem

**Review by SevenWaysDP — CHANGES_REQUESTED** [review]

File: src/serverApi/v3/api.ts [hunk 2]
```diff
@@ -2518,20 +2637,138 @@ export class TaskApi extends BaseAPI implements TaskApiInterface {
...
+         * @param {'student' | 'teacher' | 'admin'} [role] 
+         * @param {'asc' | 'desc'} [sortOrder] 
+         * @param {'firstName' | 'lastName'} [sortBy] 
          * @param {number} [skip] Number of elements (not pages) to be skipped
          * @param {number} [limit] Page limit, defaults to 10.
          * @param {*} [options] Override http request option.
          * @throws {RequiredError}
          */
-        importUserControllerFindAll: async (firstName?: string, lastName?: string, loginName?: string, match?: 'auto' | 'admin' | 'none', flagged?: boolean, skip?: number, limit?: number, options: any = {}): Promise<RequestArgs> => {
+        importUserControllerFindAllImportUsers: async (firstName?: string, lastName?: string, loginName?: string, match?: Array<'auto' | 'admin' | 'none'>, flagged?: boolean, classes?: string, role?: 'student' | 'teacher' | 'admin', sortOrder?: 'asc' | 'desc', sortBy?: 'firstName' | 'lastName', skip?: number, limit?: number, options: any = {}): Promise<RequestArgs> => {
```
Comment (SevenWaysDP) [inline src/serverApi/v3/api.ts:2771]: Many functions have been renamed. Please check!

**Review by SevenWaysDP — CHANGES_REQUESTED** [review]

File: src/components/templates/TasksDashboardStudent.vue [hunk 3]
```diff
@@ -3,6 +3,7 @@
 		<v-tabs-items v-model="currentTab">
 			<v-tab-item>
 				<v-custom-double-panels
+					class="pb-16"
```
[the remark this answers is not in the document]
Comment (odalys-dataport) [reply src/components/templates/TasksDashboardStudent.vue:6]: Yes, it is :)

File: src/serverApi/v3/api.ts (hunk 2 above)
Comment (odalys-dataport) [reply src/serverApi/v3/api.ts:2771]: Mmh I'll update the server and generate again to be sure, thanks!

**Review by SevenWaysDP — DISMISSED** [review]

**Review by SevenWaysDP — APPROVED** [review]

**Pull request merged**